src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSerializationVisitor.cs

```diff
@@ -145,7 +145,7 @@ protected override void ProcessEntityType(IEdmEntityType element)
             this.BeginElement(element, this.schemaWriter.WriteEntityTypeElementHeader);
             if (element.DeclaredKey != null && element.DeclaredKey.Any())

```

Comment:
What if `element.DeclaredKey` is null but `element.DeclareKeyAlias` is not null (i.e. the entity type key is only composed of aliases)` ? Shouldn't you still visit the key in that case?

Reply:
What is happening here is: we first visit the key then using the key name we check if that key has an alias or not.